templates/stackage-diff.hamlet

```diff
@@ -1,5 +1,10 @@
 <h1>Compare Stackage snapshots
 
+$if isJust mprevprevSnapName
+  <div .snapshot-nav>
+    <a href=@{StackageDiffR (fromJust mprevprevSnapName) name1}>
+      \< previous diff
```

Comment:
```suggestion
$maybe prevName <- mprevprevSnapName
  <div .snapshot-nav>
    <a href=@{StackageDiffR prevName name1}>
      \< previous diff
```

https://www.yesodweb.com/book/shakespearean-templates#shakespearean-templates_maybe

(If you do this, you can also remove the import of Data.Maybe.)

Reply:
Great I was looking for something like this but the shakespeare package doesn't mention that I could see.

I would commit your suggestion but `name1` _is_ the "prevSnapName", hence the awkward "prevprev" name.

(It also why I decided not to add a "newer diff" link yet, to reduce the number of queries, but let's see how this performs first...)

ps Ideally I wish the route would calculate the previous snapshot, but that would be a breaking url change too;
further we still want to be able to support diff'ing arbitrary snapshots anyway.